src/transaction/transaction.rs

```diff
@@ -271,8 +314,16 @@ impl Transaction {
         Ok(())
     }
 
-    /// Locks the given keys.
+    /// Lock the given keys without mutating value (at the time of commit).
     ///
+    /// In optimistic mode, write conflicts are not checked until commit.
+    /// So use this command to indicate that
+    /// "I do not want to commit if the value associated with this key has been modified".
+    /// It's useful to avoid *Write Skew* anomaly.
+    ///
+    /// In pessimistic mode, please use [`get_for_update`](Transaction::get_for_update) instead.
```

Comment:
Do we use this command for pessimistic transactions in TiDB?

Reply:
I think we don't. `Lock` is used for `SELECT FOR UPDATE` in optimistic transactions.

Reply:
I thought we could only select for update in pessimistic transactions, I'm probably wrong.

Reply:
Makes sense.